packages/account/src/providers/provider.ts

```diff
@@ -485,6 +485,13 @@ export default class Provider {
     } else {
       this.cache = new ResourceCache(DEFAULT_RESOURCE_CACHE_TTL);
     }
+
+    // Call auto refresh cache on 1 minute intervals continuously
+    setInterval(() => {
+      this.autoRefreshCache().catch((error) => {
+        console.error('Error during auto refresh cache:', error);
+      });
+    }, 60000); // 60000 milliseconds = 1 minute
```

Comment:
You must clear the previous interval before setting another one, otherwise it'll pile up.

Reply:
That's done in the [fetchChainAndNodeInfo](https://github.com/FuelLabs/fuels-ts/blob/mc/feat/fetch-updated-consensus-params/packages/account/src/providers/provider.ts#L631-L634) method

Reply:
I'm referring to the `setInterval` — if you create 10x `Provider` instances, you'll have 10x `setInterval` in parallel doing the same thing, which will pile up, bloat network requests, eat memory, etc. The easiest way to prevent this is to ensure you call `setInterval` only once for the lifetime of the `Provider` class.

Reply:
Also, for the FE application, it should have a way to specify if we want to have the internal, as it may be a short-lived instance of the Provider.

Also ensure we add a destroy method that allows to destroy the provider, that in this case would be mainly clean timeout.

Reply:
My apologies I misread @arboleya adjusted in d78d37bac

Good suggestion @luizstacio added in d78d37bac